apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs

```diff
@@ -535,127 +535,218 @@ public void PartitionedUpdate_Medium()
+            string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key";
+            var command = connection.CreateDmlCommand(dml);
+            command.Parameters.Add("key", SpannerDbType.String, key);
+
+            bool read = executeWithExplicitTransaction ?
+                await ExecuteWithExplicitTransaction(() => DmlAsyncWork(), connection, command) :
+                await ExecuteWithEphemeralTransaction(() => DmlAsyncWork());
+
+            async Task<bool> DmlAsyncWork()
             {
-                string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key";
-                var command = connection.CreateDmlCommand(dml);
                 using var reader = await command.ExecuteReaderAsync();
-                while (await reader.ReadAsync())
-                {
-                    // TODO: Yet to decide behaviour.
-                    //actualValues.Add(reader.GetFieldValue<int>(reader.GetOrdinal("Value")));
-                }
-            });
-            var expectedValues = new List<int> { 2, 5 };
-            Assert.Equal(expectedValues, actualValues);
+                var readOnce = await reader.ReadAsync();
+                return readOnce;
+            }
+
+            Assert.False(read);
+            AssertActualAndUpdatedDbValues(key);
         }
 
-        [Fact(Skip = "This test is skipped until DML Returning changes are done.")]
-        public async Task NoDMLReturn_ExecuteReader_NoRead()
+        // ExecuteReaderAsync still updates the database, even if we don't read from the returned reader and don't include "Then Return".
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task NoDMLReturn_ExecuteReader_NoRead(bool executeWithExplicitTransaction)
         {
             string key = _fixture.CreateTestRows();
-            var actualValues = new List<int>();
-
             using var connection = _fixture.GetConnection();
             await connection.OpenAsync();
-            RetryHelpers.ExecuteWithRetry(async () =>
+            string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key";
+            var command = connection.CreateDmlCommand(dml);
+            command.Parameters.Add("key", SpannerDbType.String, key);
+
+            _ = executeWithExplicitTransaction ?
+                await ExecuteWithExplicitTransaction(() => DmlAsyncWork(), connection, command) :
+                await ExecuteWithEphemeralTransaction(() => DmlAsyncWork());
+
+            async Task<List<int>> DmlAsyncWork()
+            {
+                var reader = await command.ExecuteReaderAsync();
+                return null;
+            }
+
+            AssertActualAndUpdatedDbValues(key);
+        }
+
+        [Fact]
+        [Trait(Constants.SupportedOnEmulator, Constants.No)]
+        public async Task DmlReturnWithRetryableTransaction()
```

Comment:
I think we can run all above tests with retryable transaction as well. If you change the "DoAsyncWork" methods to recieve a transaction, that will be null with ephemeral transactions, it should be simple I think.

Reply:
Done.